xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -586,98 +584,122 @@ void CGUIDialogAddonInfo::OnEnableDisable()
-    // Find add-on in local installation
-    CServiceBroker::GetAddonMgr().GetAddon(it.id, local_addon);
+    auto pDialog = CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+        WINDOW_DIALOG_SELECT);
+    CFileItemList items;
 
-    // All combinations of dep_addon and local_addon validity are possible and information
-    // must be displayed even when there is no dep_addon.
-    // info_addon is the add-on to take the information to display (name, icon) from. The
-    // version in the repository is preferred because it might contain more recent data.
-    info_addon = dep_addon ? dep_addon : local_addon;
-
-    if (info_addon)
+    for (const auto& it : m_depsInstalledWithAvailable)
     {
-      CFileItemPtr item(new CFileItem(info_addon->Name()));
-      std::stringstream str;
-      str << it.id << " " << it.versionMin.asString() << " -> " << it.version.asString();
-      if ((it.optional && !local_addon) || (!it.optional && local_addon))
-        str << " "
-            << StringUtils::Format(g_localizeStrings.Get(39022).c_str(),
-                                   local_addon ? g_localizeStrings.Get(39019).c_str()
-                                               : g_localizeStrings.Get(39018).c_str());
-      else if (it.optional && local_addon)
-        str << " "
-            << StringUtils::Format(g_localizeStrings.Get(39023).c_str(),
-                                   g_localizeStrings.Get(39019).c_str(),
-                                   g_localizeStrings.Get(39018).c_str());
-
-      item->SetLabel2(str.str());
-      item->SetArt("icon", info_addon->Icon());
-      item->SetProperty("addon_id", it.id);
-      items.Add(item);
-    }
-    else
-    {
-      CFileItemPtr item(new CFileItem(it.id));
-      item->SetLabel2(g_localizeStrings.Get(10005)); // Not available
-      items.Add(item);
-    }
-  }
+      // All combinations of depAddon and localAddon validity are possible and information
+      // must be displayed even when there is no depAddon.
+      // info_addon is the add-on to take the information to display (name, icon) from. The
+      // version in the repository is preferred because it might contain more recent data.
 
-  CFileItemPtr backup_item = GetCurrentListItem();
-  while (true)
-  {
-    pDialog->Reset();
-    pDialog->SetHeading(reactivate ? 39024 : 39020);
-    pDialog->SetUseDetails(true);
-    for (auto& it : items)
-      pDialog->Add(*it);
-    pDialog->EnableButton(!reactivate, 186);
-    pDialog->SetButtonFocus(true);
-    pDialog->Open();
-
-    if (pDialog->IsButtonPressed())
-      return true;
+      std::shared_ptr<IAddon> infoAddon = it.m_available ? it.m_available : it.m_installed;
+
+      if (infoAddon)
+      {
+        if (entryPoint != EntryPoint::UPDATE ||
+            (entryPoint == EntryPoint::UPDATE && !it.m_isInstalledUpToDate()))
+        {
+          const CFileItemPtr item = std::make_shared<CFileItem>(infoAddon->Name());
```

Comment:
@phunkyfish the final screenshot should just demonstrate how it looks like if there is an installed dependency that has an update available. (like `Minimum: 2.12.4 / Installed: 2.22.0.1 => Update to: 2.22.0+matrix.1`) you will not see dependencies like `requests` or `certifi` as they're filtered out as uniteresting low-level to the user.

the above logic should only list deps if we're performing a manual update and the dep has an update.
having that said i realize this could lead to an empty list being displayed.. so the dialog must not open if `CFileItemList items;` is empty.. will add this

Reply:
> the dialog must not open if `CFileItemList items;` is empty.. will add this

done with the latest push, anything else unchanged

Reply:
Cool!